src/layouts/PostDetails.astro

```diff
@@ -106,4 +106,53 @@ const layoutProps = {
     document.body.scrollTop = 0; // For Safari
     document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
   });
+
+  const copyButton = function () {
+    let copyButtonLabel = "Copy";
+    let codeBlocks = Array.from(document.querySelectorAll("pre"));
+
+    for (let codeBlock of codeBlocks) {
+      let wrapper = document.createElement("div");
+      wrapper.style.position = "relative";
+
+      let copyButton = document.createElement("button");
+      copyButton.className = "copy-code";
+      copyButton.innerHTML = copyButtonLabel;
+      copyButton.style.position = "absolute";
+      copyButton.style.top = '-10px';
+      copyButton.style.right = '10px';
+      copyButton.style.backgroundColor = "#3730a3";
```

Comment:
Also can you make the background color of the button to be a part of the theme color?

Reply:
I can co-author this if necessary.

Reply:
Yes, please. I have no experience with UI/UX, because I work mostly on the backend.